tests/user_examples/extra_src/shared/general_continuum/continuum_dynamics/continuum_dynamics_complex.hpp

```diff
@@ -70,5 +70,114 @@ void BaseShearStressRelaxation2ndHalfWithWall<BaseShearStressRelaxation2ndHalfTy
+{
+    return this->acc_prior_[index_i];
+}
+
+template <class BaseStressRelaxation1stHalfType>
+void BaseStressRelaxation1stHalfWithWall<BaseStressRelaxation1stHalfType>::interaction(size_t index_i, Real dt)
+{
+    BaseStressRelaxation1stHalfType::interaction(index_i, dt);
+
+    Vecd acc_prior_i = computeNonConservativeAcceleration(index_i);
+    Vecd acceleration = acc_prior_i;
+    Real rho_dissipation(0);
+
+    Matd stress_tensor_i = this->reduceTensor(this->stress_tensor_3D_[index_i]);
+
+    for (size_t k = 0; k < fluid_dynamics::FluidWallData::contact_configuration_.size(); ++k)
+    {
+        StdLargeVec<Vecd>& acc_ave_k = *(this->wall_acc_ave_[k]);
+        Neighborhood& wall_neighborhood = (*fluid_dynamics::FluidWallData::contact_configuration_[k])[index_i];
+        for (size_t n = 0; n != wall_neighborhood.current_size_; ++n)
+        {
+            size_t index_j = wall_neighborhood.j_[n];
+            Vecd& e_ij = wall_neighborhood.e_ij_[n];
+            Real dW_ijV_j = wall_neighborhood.dW_ijV_j_[n];
+            Real r_ij = wall_neighborhood.r_ij_[n];
+
+            Real face_wall_external_acceleration = (acc_prior_i - acc_ave_k[index_j]).dot(-e_ij);
+            Real p_in_wall = this->p_[index_i] + this->rho_[index_i] * r_ij * SMAX(Real(0), face_wall_external_acceleration);
+            acceleration += 2 * stress_tensor_i * wall_neighborhood.dW_ijV_j_[n] * wall_neighborhood.e_ij_[n];
+            rho_dissipation += this->riemann_solver_.DissipativeUJump(this->p_[index_i] - p_in_wall) * dW_ijV_j;
+        }
+    }
+    this->acc_[index_i] += acceleration / this->rho_[index_i];
+    this->drho_dt_[index_i] += rho_dissipation * this->rho_[index_i];
+}
+//=================================================================================================//
+template <class BaseStressRelaxation2ndHalfType>
+void BaseStressRelaxation2ndHalfWithWall<BaseStressRelaxation2ndHalfType>::interaction(size_t index_i, Real dt)
+{
+    BaseStressRelaxation2ndHalfType::interaction(index_i, dt);
+
+    Real density_change_rate = 0.0;
+    Vecd p_dissipation = Vecd::Zero();
+
+    Vecd vel_i = this->vel_[index_i];
+    Matd velocity_gradient = Matd::Zero();
+
+    for (size_t k = 0; k < fluid_dynamics::FluidWallData::contact_configuration_.size(); ++k)
+    {
+        StdLargeVec<Vecd>& vel_ave_k = *(this->wall_vel_ave_[k]);
+        StdLargeVec<Vecd>& n_k = *(this->wall_n_[k]);
+
+        Neighborhood& wall_neighborhood = (*fluid_dynamics::FluidWallData::contact_configuration_[k])[index_i];
+        for (size_t n = 0; n != wall_neighborhood.current_size_; ++n)
+        {
+            size_t index_j = wall_neighborhood.j_[n];
+            Vecd& e_ij = wall_neighborhood.e_ij_[n];
+            Real dW_ijV_j = wall_neighborhood.dW_ijV_j_[n];
+            Vecd nablaW_ijV_j = wall_neighborhood.dW_ijV_j_[n] * wall_neighborhood.e_ij_[n];
+            Vecd vel_in_wall = 1.9 * vel_ave_k[index_j] - 0.9 *vel_i;
```

Comment:
the constant parameters 1.9 and 0.9 is general ?

Reply:
The reference paper gives a range for the parameter, and the specific value is selected according to numerical experiments and generally used for all cases.